File: src/main/java/ru/quandastudio/lpsserver/validation/BanlistValidator.java
package ru.quandastudio.lpsserver.validation;

import javax.validation.ConstraintValidator;
import javax.validation.ConstraintValidatorContext;

import ru.quandastudio.lpsserver.core.Banlist;

public class BanlistValidator implements ConstraintValidator<CheckInBanlist, String> {

	@Override
	public boolean isValid(String value, ConstraintValidatorContext context) {
		return value != null && Banlist.checkLogin(value);
	}

}

File: src/main/java/ru/quandastudio/lpsserver/validation/BanlistValidator.kt
package ru.quandastudio.lpsserver.validation

import ru.quandastudio.lpsserver.core.ContextProvider
import ru.quandastudio.lpsserver.data.ForbiddenLoginManager
import javax.validation.ConstraintValidator
import javax.validation.ConstraintValidatorContext

class BanlistValidator : ConstraintValidator<CheckInBanlist, String?> {

    override fun initialize(constraintAnnotation: CheckInBanlist?) {
        super.initialize(constraintAnnotation)
    }

    override fun isValid(value: String?, context: ConstraintValidatorContext): Boolean {
        val forbiddenLoginManager = ContextProvider.getBean(ForbiddenLoginManager::class.java)

        return value == null || forbiddenLoginManager.isLoginValid(value)
    }
}